openpgp/s2k/s2k.go

```diff
@@ -14,56 +15,123 @@ import (
 
 	"github.com/ProtonMail/go-crypto/openpgp/errors"
 	"github.com/ProtonMail/go-crypto/openpgp/internal/algorithm"
+	"golang.org/x/crypto/argon2"
 )
 
-// Config collects configuration parameters for s2k key-stretching
-// transformations. A nil *Config is valid and results in all default
-// values. Currently, Config is used only by the Serialize function in
-// this package.
-type Config struct {
-	// S2KMode is the mode of s2k function.
-	// It can be 0 (simple), 1(salted), 3(iterated)
+type S2KType uint8
+
+// Defines the default S2KMode constants
+//
+//	0 (simple), 1(salted), 3(iterated), 4(argon2)
+const (
+	SimpleS2K     S2KType = 0
+	SaltedS2K             = 1
+	IterSaltedS2K         = 3
+	Argon2S2K             = 4
+	GnuS2K                = 101
+)
+
+const Argon2S2KDefaultNonceSize int = 16
+
+// S2KConfig collects configuration parameters for s2k key-stretching
+// transformations. A nil *S2KConfig is valid and results in all default
+// values.
+type S2KConfig struct {
+	// S2K (String to Key) mode, used for key derivation in the context of secret key encryption
+	// and password-encrypted data. Either s2k.Argon2S2K or s2k.IterSaltedS2K has to be selected
+	// weaker options are not allowed.
+	// Note: Argon2 is the strongest option but not all OpenPGP implementations are compatible with it
+	//(pending standardisation).
+	// 0 (simple), 1(salted), 3(iterated), 4(argon2)
 	// 2(reserved) 100-110(private/experimental).
-	S2KMode uint8
+	S2KMode S2KType
+	// Only relevant if S2KMode is not set to s2k.Argon2S2K.
 	// Hash is the default hash function to be used. If
 	// nil, SHA256 is used.
 	Hash crypto.Hash
-	// S2KCount is only used for symmetric encryption. It
+	// Argon2 parameters for S2K (String to Key).
+	// Only relevant if S2KMode is set to s2k.Argon2S2K.
+	// If nil, default parameters are used.
+	// For more details on the choice of parameters, see https://tools.ietf.org/html/rfc9106#section-4.
+	ArgonConf *ArgonConfig
```

Comment:
And similarly, I'd call this `Argon2Config` (both the field and the type)

Reply:
(this one didn't quite get resolved, note the `2` :slightly_smiling_face: since we refer to it by the full name everywhere else I would do so for the config field and type name as well)